yasnippet.el

```diff
@@ -3035,18 +3035,30 @@ through the field's start point"
    (not (and (yas--field-number field)
              (zerop (yas--field-number field))))))
 
-(defun yas--snippets-at-point (&optional all-snippets)
-  "Return a sorted list of snippets at point.
-
-The most recently-inserted snippets are returned first."
-  (sort
-   (delq nil (delete-dups
-              (mapcar (lambda (ov) (overlay-get ov 'yas--snippet))
-                      (if all-snippets (overlays-in (point-min) (point-max))
-                        (nconc (overlays-at (point))
-                               (overlays-at (1- (point))))))))
-   #'(lambda (s1 s2)
-       (<= (yas--snippet-id s2) (yas--snippet-id s1)))))
+(defun yas-active-snippets (&optional where)
+  "Return a sorted list of active snippets.
+The most recently-inserted snippets are returned first.
+
+By default, lists snippets around point.  If optional argument
```

Comment:
> I'd prefer a protocol matching overlays-at or overlays-in.

You mean `(defun yas-active-snippets (&optional begin end)` instead of requiring `cons`ing?

> clarify that in the `consp` case you use `overlays-in` (not really "within the given range")

I'm not sure if the docstring should really be committing to a particular implementation.